pkg/common/common.go

```diff
@@ -23,3 +23,43 @@ const (
 	// Resource lifecycle annotation value to prevent deletion.
 	OnRemoveKeep = "keep"
 )
+
+var Strategies = []DryRunStrategy{DryRunClient, DryRunServer}
+
+type DryRunStrategy int
+
+const (
+	// DryRunNone indicates the client will make all mutating calls
+	DryRunNone DryRunStrategy = iota
+
+	// DryRunClient, or client-side dry-run, indicates the client will prevent
+	// making mutating calls such as CREATE, PATCH, and DELETE
+	DryRunClient
+
+	// DryRunServer, or server-side dry-run, indicates the client will send
+	// mutating calls to the APIServer with the dry-run parameter to prevent
+	// persisting changes.
+	//
+	// Note that clients sending server-side dry-run calls should verify that
+	// the APIServer and the resource supports server-side dry-run, and otherwise
+	// clients should fail early.
+	//
+	// If a client sends a server-side dry-run call to an APIServer that doesn't
+	// support server-side dry-run, then the APIServer will persist changes inadvertently.
+	DryRunServer
+)
+
+// ClientDryRun returns true if input drs is DryRunClient
+func ClientDryRun(drs DryRunStrategy) bool {
+	return drs == DryRunClient
+}
+
+// ServerDryRun returns true if input drs is DryRunServer
+func ServerDryRun(drs DryRunStrategy) bool {
+	return drs == DryRunServer
+}
+
+// DryRun returns true if input drs is any of the dry runs
+func DryRun(drs DryRunStrategy) bool {
```

Comment:
Could we use a name here that makes it clearer what this function actually does?

Reply:
Done.